### PR TITLE
cleanup: Use Go 1.19 atomic types

### PR DESCRIPTION
**Pull request by ohill**
## Summary

Go 1.19 introduced types like atomic.Uint64, atomic.Int32, atomic.Bool, that provide automatic alignment for 32-bit platforms, plus extra safety to ensure access to the underlying value only uses atomic methods. This converts nearly all the calls to atomic to use these types. It also catches a few places where non-atomic and atomic access were mixed together.

## Test Plan

Removed TestAtomicVariablesAlignment

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -18,7 +18,6 @@ linters:
     - revive
     - staticcheck
     - typecheck
-    - paralleltest
```
Comment (jasonpaulos) [inline .golangci.yml:21]: Why get rid of this linter?

File: data/pools/transactionPool.go [hunk 2]
```diff
@@ -65,6 +62,7 @@ type TransactionPool struct {
 	expiredTxCount         map[basics.Round]int
 	pendingBlockEvaluator  BlockEvaluator
 	numPendingWholeBlocks  basics.Round
+	feePerByte             atomic.Uint64
```
Comment (jasonpaulos) [inline data/pools/transactionPool.go:65]: Just wanted to add context by mentioning that according to https://pkg.go.dev/sync/atomic#pkg-note-BUG, types atomic.Int64 and atomic.Uint64 are automatically 64-bit aligned, so they can be placed anywhere in a struct definition

File: ledger/bulletin.go [hunk 3]
```diff
@@ -31,17 +31,17 @@ import (
 // notifier is a struct that encapsulates a single-shot channel; it will only be signaled once.
 type notifier struct {
 	signal   chan struct{}
-	notified uint32
+	notified *atomic.Uint32
```
Comment (jasonpaulos) [inline ledger/bulletin.go:34]: Why is this one a pointer when all other replacements have not been?

File: rpcs/blockService.go [hunk 4]
```diff
@@ -320,9 +320,9 @@ func (bs *BlockService) handleCatchupReq(ctx context.Context, reqMsg network.Inc
 		outMsg := network.OutgoingMessage{Topics: respTopics}
 		if n > 0 {
 			outMsg.OnRelease = func() {
-				atomic.AddUint64(&bs.wsMemoryUsed, ^uint64(n-1))
+				bs.wsMemoryUsed.Add(^uint64(n - 1))
 			}
-			atomic.AddUint64(&bs.wsMemoryUsed, (n))
+			bs.wsMemoryUsed.Add((n))
```
Comment (jasonpaulos) [inline rpcs/blockService.go:325]: nit: unnecessary parenthesis

**Review by algorandskiy — COMMENTED** [review]
Basically the same questions as Jason had

File: ledger/bulletin.go (hunk 3 above)
Comment (jasonpaulos) [reply ledger/bulletin.go:34]: Ah, that is subtle. Makes sense.

**Review by jasonpaulos — APPROVED** [review]

**Review by algorandskiy — APPROVED** [review]

File: ledger/eval/prefetcher/prefetcher.go [hunk 5]
```diff
@@ -385,21 +382,22 @@ func (p *accountPrefetcher) prefetch(ctx context.Context) {
...
 		gr.balances = allBalances[usedBalances : usedBalances+gr.balancesCount]
 		if gr.resourcesCount > 0 {
 			gr.resources = allResources[usedResources : usedResources+gr.resourcesCount]
 			usedResources += gr.resourcesCount
 		}
 		usedBalances += gr.balancesCount
-		if gr.incompleteCount == 0 {
-			gr.incompleteCount = dependencyFreeGroup
+		if gr.incompleteCount.Load() == 0 {
+			gr.incompleteCount.Store(dependencyFreeGroup)
```
[the remark this answers is not in the document]
Comment (algorandskiy) [reply ledger/eval/prefetcher/prefetcher.go:394]: This is correct, non-atomic direct access is part of initialization.

File: shared/pingpong/pingpong.go [hunk 6]
```diff
@@ -1036,11 +1036,11 @@ type paymentUpdate struct {
 }
 
 func (au *paymentUpdate) apply(pps *WorkerState) {
-	pps.accounts[au.from].balance -= (au.fee + au.amt)
+	pps.accounts[au.from].balance.Add(-(au.fee + au.amt))
```
[the remark this answers is not in the document]
Comment (algorandskiy) [reply shared/pingpong/pingpong.go:1039]: pingpong's main logic is single-threaded

File: network/wsNetwork.go [hunk 7]
```diff
@@ -2466,14 +2466,14 @@ func (wn *WebsocketNetwork) postMessagesOfInterestThread() {
 		<-wn.messagesOfInterestRefresh
 		// if we're not a relay, and not participating, we don't need txn pool
 		wantTXGossip := wn.nodeInfo.IsParticipating()
-		if wantTXGossip && (wn.wantTXGossip != wantTXGossipYes) {
+		if wantTXGossip && (wn.wantTXGossip.Load() != wantTXGossipYes) {
```
[the remark this answers is not in the document]
Comment (algorandskiy) [reply network/wsNetwork.go:2469]: right, the same pattern is before - set in init, modify/read here. It is probably even OK not to be atomic.

[the remark this answers is not in the document]
Comment (algorandskiy) [reply network/wsNetwork.go:2469]: yep, I roughly remember when this became atomic b/c of a data race in tests.

**Pull request merged**